File: src/bit_rotation.c
#include <stdio.h>

#define LEFT 0
#define RIGHT 1
#define BIT_INTS 32

typedef unsigned short int t_bool;

int bit_rotation(t_bool, int, unsigned int);

int main() {
    int i = 1;
    int j;

    printf("Initial value: %d\n", i);

    for (j=0; j<3; j++) {
        i = bit_rotation(LEFT, i, 1);

        printf("Iter: %d\n\tValue: %d\n", j+1, i);
    }

    return 0;
}

int bit_rotation(t_bool mode, int input_num, unsigned int bits_to_rotate) {
    if (mode == 0) {
        return (input_num << bits_to_rotate) | (input_num >> (BIT_INTS - bits_to_rotate));
    } else {
        return (input_num >> bits_to_rotate) | (input_num << (BIT_INTS - bits_to_rotate));
    }
}

File: src/bin/bit_rotation.rs
pub mod bit_rotation {
    pub const INT_BITS: i32 = 32;

    pub fn left(i: i32, j: i32) -> i32 {
        (i << j) | (i >> (INT_BITS - j))
    }

    pub fn left2(i: i32, j: i32) -> i32 {
        (i << (4 * j)) | (i >> (INT_BITS - j))
    }

    pub fn right(i: i32, j: i32) -> i32 {
        (i >> j) | (i << (INT_BITS - j))
    }
}

fn main() {
    let mut i = 1;

    println!("Init Value: {}", i);

    for _ in 0..3 {
        i = bit_rotation::left2(i, 1);

        println!("Value: {}", i);
    }
}
